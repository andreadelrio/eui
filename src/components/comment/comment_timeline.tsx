import React, { FunctionComponent, ReactNode } from 'react';
import { CommonProps, keysOf } from '../common';
import classNames from 'classnames';
import { EuiIcon } from '../icon';

export type EuiCommentTimelineProps = CommonProps & {
  /**
   * Main icon that accompanies the comment. There's default icons for both types of comment, user icon for regular comments and dot icon for update comments. It also accepts any type supported by EuiIcon or any other node.
   */
  timelineIcon?: ReactNode | string;
  type?: EuiCommentType;
};

const typeToClassNameMap = {
  regular: 'euiCommentTimeline--regular',
  update: 'euiCommentTimeline--update',
};

export const TYPES = keysOf(typeToClassNameMap);
export type EuiCommentType = keyof typeof typeToClassNameMap;

export const EuiCommentTimeline: FunctionComponent<EuiCommentTimelineProps> = ({
  className,
  timelineIcon,
  type = 'regular',
  ...rest
}) => {
  const classes = classNames('euiCommentTimeline', className);
  const iconClasses = classNames(
    {
      euiCommentTimeline__default:
        !timelineIcon || typeof timelineIcon === 'string',
    },
    typeToClassNameMap[type]
  );

  let iconRender;
  if (typeof timelineIcon === 'string') {
    iconRender = (
      <EuiIcon size={type === 'update' ? 'm' : 'l'} type={timelineIcon} />
    );
  } else if (timelineIcon) {
    iconRender = timelineIcon;
  } else {
    iconRender = (
      <EuiIcon
        type={type === 'update' ? 'dot' : 'user'}
        size={type === 'update' ? 's' : 'l'}
      />
    );
  }

  return (
    <div className={classes} {...rest}>
      <div className="euiCommentTimeline__content">
        <div className={iconClasses}>{iconRender}</div>
      </div>
    </div>
  );
};
