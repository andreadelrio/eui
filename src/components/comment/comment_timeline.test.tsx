import React from 'react';
import { render } from 'enzyme';
import { requiredProps } from '../../test/required_props';

import { EuiCommentTimeline } from './comment_timeline';

describe('EuiCommentTimeline', () => {
  test('is rendered', () => {
    const component = render(<EuiCommentTimeline {...requiredProps} />);

    expect(component).toMatchSnapshot();
  });
});
