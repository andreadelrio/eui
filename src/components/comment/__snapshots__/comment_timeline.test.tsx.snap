// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EuiCommentTimeline is rendered 1`] = `
<div
  aria-label="aria-label"
  class="euiCommentTimeline testClass1 testClass2"
  data-test-subj="test subject string"
>
  <div
    class="euiCommentTimeline__content"
  >
    <div
      class="euiCommentTimeline__icon--default euiCommentTimeline__icon--regular"
    >
      <div
        data-euiicon-type="user"
      />
    </div>
  </div>
</div>
`;
